case/src/test/kotlin/com/ritense/case_/service/CaseWidgetTabServiceIntTest.kt

```diff
@@ -66,4 +69,150 @@ class CaseWidgetTabServiceIntTest @Autowired constructor(
         assertThat(caseWidgetTabRepository.existsById(tabId)).isFalse()
     }
 
+    @Test
+    fun `should add widgets to widget tab`() {
+        val caseDefinitionName = "some-case-type"
+        val tabKey = "my-tab"
+
+        runWithoutAuthorization {
+            caseTabService.createCaseTab(
+                caseDefinitionName,
+                CaseTabDto(key = tabKey, type = CaseTabType.WIDGETS, contentKey = "-")
+            )
+
+            caseWidgetTabService.updateWidgetTab(
+                CaseWidgetTabDto(
+                    caseDefinitionName,
+                    tabKey,
+                    widgets = listOf(
+                        CaseWidgetTabWidgetDto("widget-1", "Widget 1", 0, false),
+                        CaseWidgetTabWidgetDto("widget-2", "Widget 2", 1, true)
+                    )
+                )
+            )
+        }
+
+        val optionalWidgetTab = caseWidgetTabRepository.findById(CaseTabId(caseDefinitionName, tabKey))
+
+        assertTrue(optionalWidgetTab.isPresent)
+
+        val widgetTab = optionalWidgetTab.get()
+
+        assertThat(widgetTab.widgets).hasSize(2)
+        assertThat(widgetTab.widgets[0].key).isEqualTo("widget-1")
+        assertThat(widgetTab.widgets[1].key).isEqualTo("widget-2")
+        assertThat(widgetTab.widgets[0].title).isEqualTo("Widget 1")
+        assertThat(widgetTab.widgets[1].title).isEqualTo("Widget 2")
+        assertThat(widgetTab.widgets[0].width).isEqualTo(0)
+        assertThat(widgetTab.widgets[1].width).isEqualTo(1)
+        assertThat(widgetTab.widgets[0].highContrast).isFalse()
+        assertThat(widgetTab.widgets[1].highContrast).isTrue()
+        assertThat(widgetTab.widgets[0].order).isEqualTo(0)
+        assertThat(widgetTab.widgets[1].order).isEqualTo(1)
+    }
+
+    @Test
+    fun `should remove widgets from widget tab`() {
+        val caseDefinitionName = "some-case-type"
+        val tabKey = "my-tab"
+
+        runWithoutAuthorization {
+            caseTabService.createCaseTab(
+                caseDefinitionName,
+                CaseTabDto(key = tabKey, type = CaseTabType.WIDGETS, contentKey = "-")
+            )
+
+            caseWidgetTabService.updateWidgetTab(
+                CaseWidgetTabDto(
+                    caseDefinitionName,
+                    tabKey,
+                    widgets = listOf(
+                        CaseWidgetTabWidgetDto("widget-1", "Widget 1", 0, false),
+                        CaseWidgetTabWidgetDto("widget-2", "Widget 2", 1, true)
+                    )
+                )
+            )
+        }
+
+        val optionalWidgetTab = caseWidgetTabRepository.findById(CaseTabId(caseDefinitionName, tabKey))
```

Comment:
This would be way nicer if you used `findByIdOrNull` instead. Same for all other usages of `.findById` in this file.